benchmarking/tuning.py

```diff
@@ -0,0 +1,174 @@
+"""Tunes the hyperparameters of the algorithms."""
+
+import copy
+import pathlib
+from typing import Any, Dict
+
+import numpy as np
+import ray
+from pandas.api import types as pd_types
+from ray.tune.search import optuna
+from sacred.observers import FileStorageObserver
+from tuning_config import parallel_ex, tuning_ex
```

Comment:
PEP8: standard library, third-party then first party-imports with line separating each.
```suggestion

from tuning_config import parallel_ex, tuning_ex
```